### PR TITLE
Make working docker arm64 image

### PR DESCRIPTION
**Pull request by lukas-phaf**
Mostly the changes are around getting a working docker arm64 image, both locally and on GitHub. This involved loosening the definition of when an output picture is "the same".

I disabled parallel builds, as the default parallel setting make the GitHub action runner run out of memory for the (emulated) arm64 build. I still need to look at what the best settings are. But although there might be slight improvements, build times will be extremely long (1 hour). We need to decide on how we want to handle this.

There are also other changes, which are suggested improvements. Among others:

- Invert `.dockerignore` setup
- Disable gcc C++ extensions

### CODE REVIEW

File: doc/developing/scripts/mac_install_dependencies.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ pyenv virtualenv 3.9.13 adaguc
 pyenv activate adaguc
 
 # Install the required Python packages
-pip install flask flask-cors flask-caching gunicorn pytest marshmallow owslib pyproj==2.6.1 apispec apispec-webframeworks marshmallow-oneofschema defusedxml netcdf4
+pip install flask flask-cors flask-caching gunicorn pytest marshmallow owslib pyproj==2.6.1 apispec apispec-webframeworks marshmallow-oneofschema defusedxml netcdf4 chardet lxml
```
Comment (maartenplieger) [inline doc/developing/scripts/mac_install_dependencies.md]: Would it be possible to use the same requirements.txt file?

File: python/lib/adaguc/AdagucTestTools.py [hunk 2]
```diff
@@ -122,6 +123,62 @@ def mkdir_p(self, directory):
         if not os.path.exists(directory):
             os.makedirs(directory)
 
+    def compareImage(self, expectedImagePath, returnedImagePath, maxAllowedColorDifference = 8):
```
Comment (maartenplieger) [inline python/lib/adaguc/AdagucTestTools.py]: Nice!

File: adagucserverEC/CConvertADAGUCVector.cpp [hunk 3]
```diff
@@ -453,8 +453,10 @@ int CConvertADAGUCVector::convertADAGUCVectorData(CDataSource *dataSource, int m
           if (projectionRequired) {
             if (imageWarper.reprojfromLatLon(lons[j], lats[j]) != 0) projectionIsOk = false;
           }
-          dlons[j] = int((lons[j] - offsetX) / cellSizeX);
-          dlats[j] = int((lats[j] - offsetY) / cellSizeY);
+          if (cellSizeX != 0 && cellSizeY != 0) {
+            dlons[j] = int((lons[j] - offsetX) / cellSizeX);
+            dlats[j] = int((lats[j] - offsetY) / cellSizeY);
+          }
```
Comment (lukas-phaf) [inline adagucserverEC/CConvertADAGUCVector.cpp]: If this is not executed, dlons[j] and dlats[j] are used uninitialised.

File: adagucserverEC/CConvertADAGUCPoint.cpp [hunk 4]
```diff
@@ -870,7 +870,7 @@ int CConvertADAGUCPoint::convertADAGUCPointData(CDataSource *dataSource, int mod
         if (discRadiusX < 0.1) discRadiusX = 0.1;
       }
 
-      if (projectionError == false) {
+      if (projectionError == false && cellSizeX != 0 && cellSizeY != 0) {
```
Comment (lukas-phaf) [inline adagucserverEC/CConvertADAGUCPoint.cpp:873]: Is it okay to do nothing if they are not set?

File: adagucserverEC/CConvertGeoJSON.cpp [hunk 5]
```diff
@@ -1685,7 +1688,7 @@ void CConvertGeoJSON::drawPoints(Feature *feature, unsigned short int featureInd
       if (pointGridVariable->getType() == CDF_FLOAT) {
         if (f < min || min != min) min = f;
         if (f > max || max != max) max = f;
-        if (pointGridVariable->data != NULL) {
+        if (!(pointGridVariable->data == nullptr)) {
```
Comment (lukas-phaf) [inline adagucserverEC/CConvertGeoJSON.cpp]:
```suggestion
        if (pointGridVariable->data != nullptr) {
```

File: adagucserverEC/CGenericDataWarper.h [hunk 6]
```diff
@@ -447,8 +447,9 @@ class GenericDataWarper {
       for (size_t y = 0; y < dataHeightStrided; y++) {
         for (size_t x = 0; x < dataWidthStrided; x++) {
           size_t pS = x + y * dataWidthStrided;
-          double valX = dfSourcedExtW * double(double(x * projStrideFactor + halfCell) + PXExtentBasedOnSource[0]) + dfSourceOrigX;
-          double valY = dfSourcedExtH * double(double(y * projStrideFactor - halfCell) + PXExtentBasedOnSource[1]) + dfSourceOrigY;
+
+          double valX = dfSourcedExtW * double(double(double(x * projStrideFactor) + halfCell) + double(PXExtentBasedOnSource[0])) + dfSourceOrigX;
+          double valY = dfSourcedExtH * double(double(double(y * projStrideFactor) - halfCell) + double(PXExtentBasedOnSource[1])) + dfSourceOrigY;
```
Comment (lukas-phaf) [inline adagucserverEC/CGenericDataWarper.h]: I don't think any of these double casts do anything?

File: doc/developing/scripts/mac_install_dependencies.md (hunk 1 above)
Comment (lukas-phaf) [reply doc/developing/scripts/mac_install_dependencies.md]: Yes... I originally missed it...

**Review by maartenplieger — APPROVED** [review]

**Pull request merged**